misc/python/materialize/checks/scenarios_upgrade.py

```diff
@@ -509,3 +510,69 @@ def actions(self) -> list[Action]:
             ),
             Validate(self),
         ]
+
+
+class SelfManagedv25_2_Upgrade(Scenario):
+    """
+    Upgrade from the oldest v25.2 patch release to the latest v25.2 patch release.
```

Comment:
We also have to upgrade to the current main state after that. We don't want to have tests which test an old state of Materialize. This test is supposed to got from v25.2.0 -> v25.2.1 -> ... -> v25.2.18 (or whatever the latest is) -> main